app-backend/batch/src/main/scala/com/azavea/rf/batch/ast/package.scala

```diff
@@ -61,44 +62,66 @@ package object ast {
 
     /* Guarantee correctness before performing Map Algebra */
     val pure = Interpreter.interpretPure[Unit](ast, sourceMapping)
-    val rdds = sourceMapping.mapValues(r => fetch(r, zoom, ingestLocs)).sequence
+    val rdds = sourceMapping.mapValues(r => fetch(r, zoom, sceneLocs, projLocs)).sequence
 
     (pure |@| rdds).map({ case (_, rs) => eval(ast, rs) })
   }
 
   /** This requires that for each [[RFMLRaster]] that a band number be specified. */
-  def fetch(
+  private def fetch(
     raster: RFMLRaster,
     zoom: Int,
-    ingestLocs: Map[UUID, String]
+    sceneLocs: Map[UUID, String],
+    projLocs: Map[UUID, List[(UUID, String)]]
   )(implicit sc: SparkContext): Interpreted[TileLayerRDD[SpatialKey]] = raster match {
 
-    case ProjectRaster(id, _) => Invalid(NonEmptyList.of(UnhandledCase(id)))
+    case ProjectRaster(id, None) => Invalid(NonEmptyList.of(NoBandGiven(id)))
+    case ProjectRaster(id, Some(band)) => getStores(id, projLocs) match {
+      case None => Invalid(NonEmptyList.of(AttributeStoreFetchError(id)))
+      case Some(stores) => {
+        val rdds: List[TileLayerRDD[SpatialKey]] =
+          stores.map({ case (sceneId, store) =>
+            S3LayerReader(store)
+              .read[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](LayerId(sceneId.toString, zoom))
+              .withContext(rdd => rdd.mapValues(_.band(band)))
+          })
+
+        Valid(rdds.reduce(_ merge _))
```

Comment:
List itself and RDDs would be in the order you read everything => you only need to be sure that the `stores` list keeps the correct ordering. I am not sure about it, as you used `Map` for `sceneLocs` and `projLocs`, can you double check that the ordering is the same during this function call?